docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md

```diff
@@ -0,0 +1,599 @@
+        // highlight-end
+        return (
+          <Helmet
+            htmlAttributes={{
+              lang: "en",
+            }}
+            title={title}
+            meta={
+              [
+                {
+                  name: "description",
+                  content: metaDescription,
+                },
+                {
+                  name: "keywords",
+                  content: data.site.siteMetadata.keywords.join(","),
+                },
+                // highlight-start
+                {
+                  property: "og:title",
+                  content: title,
+                },
+                {
+                  property: "og:description",
+                  content: metaDescription,
+                },
+                {
+                  name: "twitter:creator",
+                  content: data.site.siteMetadata.author,
+                },
+                {
+                  name: "twitter:title",
+                  content: title,
+                },
+                {
+                  name: "twitter:description",
+                  content: metaDescription,
+                },
+              ]
+                .concat(
+                  metaImage
+                    ? [
+                        {
+                          property: "og:image",
+                          content: image,
+                        },
+                        {
+                          property: "og:image:width",
+                          content: metaImage.width,
+                        },
+                        {
+                          property: "og:image:height",
+                          content: metaImage.height,
+                        },
+                        {
+                          name: "twitter:card",
+                          content: "summary_large_image",
+                        },
+                      ]
+                    : [
+                        {
+                          name: "twitter:card",
+                          content: "summary",
+                        },
+                      ]
+                )
+                .concat(meta)
+              // highlight-end
+            }
+          />
+        )
+      }}
+    />
+  )
+}
+
+// highlight-start
+SEO.defaultProps = {
+  meta: [],
+}
+// highlight-end
+
+// highlight-start
+SEO.propTypes = {
+  description: PropTypes.string,
+  image: PropTypes.shape({
+    src: PropTypes.string.isRequired(),
+    height: PropTypes.string.isRequired(),
+    width: PropTypes.string.isRequired(),
+  }),
+  meta: PropTypes.array,
+  title: PropTypes.string.isRequired,
+}
+// highlight-end
+
+export default SEO
+```
+
+Woo hoo! You enabled not only SEO for search engines like Google, Bing (people use Bing, right?) but you also laid the groundwork for enhanced sharing capabilities on social networks. Finally, you will learn to add support for one of the more useful functionalities for SEO, specifically a canonical link.
```

Comment:
```suggestion
Woo hoo! You enabled not only SEO for search engines like Google and Bing, but you also laid the groundwork for enhanced sharing capabilities on social networks. Finally, you will learn to add support for one of the more useful functionalities for SEO, specifically a canonical link.
```

I feel like the side comment makes more sense in a blog post but not so much a tutorial.